demo/src/API.elm

```diff
@@ -25,56 +26,54 @@ main =
         { configDecoder = configDecoder
         , requestPort = requestPort
         , responsePort = responsePort
-        , endpoint = Endpoint
+        , endpoint = Endpoint -- Requests will come in with this message
         , initialModel = Model []
         , parseRoute = \path -> UrlParser.parse Route.route path Dict.empty
-        , pipeline = mainPipeline
+        , update = update
         , subscriptions = subscriptions
         }
 
 
-{-| Your pipeline.
+{-| The application update function.
 
-A pipeline is a sequence of plugs, each of which transforms the connection
-in some way.
+Just like an Elm SPA, an elm-serverless app has a single update
+function which is the first point of contact for incoming messages.
 
 -}
-mainPipeline : Plug
-mainPipeline =
-    pipeline
-        -- Simple plugs just transform the connection.
-        -- A router takes a `Conn` and returns a new pipeline.
-        |> fork router
+update : Msg -> Conn -> ( Conn, Cmd Msg )
+update msg conn =
+    case msg of
+        -- New requests come in here
+        Endpoint ->
+            conn
+                |> cors
+                |> router
```

Comment:
Simple pipelines can be achieved by just using the `|>` or `>>` operators and chaining `Conn -> Conn` methods together. This makes pipelines less versatile. For example, a pipeline function can no longer issue a command, but it greater reduces the complexity of the framework